pyodide_build/recipe/builder.py

```diff
@@ -332,6 +341,7 @@ def _download_and_extract(self) -> None:
 
         # already built
         if tarballpath.suffix == ".whl":
+            check_versions_match(tarballpath.name, self.version)
```

Comment:
If we don't want to download the wheel file, I think it's possible to check before it is downloaded by looking at the URL.